### PR TITLE
Enable window-group-limit optimization on [databricks]

### LINKED ISSUE #10531: [FEA] Support "WindowGroupLimit" optimization on GPU for Databricks 13.3 ML LTS+
**Is your feature request related to a problem? Please describe.**
Issue #8208 enabled WindowGroupLimit in the Spark 3.5.0 and 3.5.1 shims.  It did not enable for Databricks 13.3 ML LTS.  

**Describe the solution you'd like**
Enable WindowGroupLimit in the Databricks 13.3 ML LTS shim.  

**Describe alternatives you've considered**

**Additional context**
https://docs.databricks.com/en/release-notes/runtime/13.3lts.html

### PR DESCRIPTION
**Pull request by mythrocks**
Fixes #10531.

This is a followup to #10500, which added support to push down window-group-limit filters before the shuffle phase.

#10500 inadvertently neglected to ensure that the optimization works on Databricks. (It turns out that window-group-limit was cherry-picked into Databricks 13.3, despite the nominal Spark version being `3.4.1`.)

This change ensures that the same optimization is available on Databricks 13.3 (and beyond).

### CODE REVIEW

**Comment by mythrocks** [comment]
Build

File: sql-plugin/src/main/spark341db/scala/com/nvidia/spark/rapids/shims/GpuWindowGroupLimitExec.scala [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*** spark-rapids-shim-json-lines
+{"spark": "341db"}
```
Comment (revans2) [inline sql-plugin/src/main/spark341db/scala/com/nvidia/spark/rapids/shims/GpuWindowGroupLimitExec.scala:18]: This and the 350 version are identical. The other file should be deleted and the shims for 350 and 351 should be added here.

Comment (mythrocks) [reply sql-plugin/src/main/spark341db/scala/com/nvidia/spark/rapids/shims/GpuWindowGroupLimitExec.scala:18]: I didn't realize that was possible.  This shim is in the databricks directory.

Comment (mythrocks) [reply sql-plugin/src/main/spark341db/scala/com/nvidia/spark/rapids/shims/GpuWindowGroupLimitExec.scala:18]: Testing this out now. Shimplify FTW.

**Comment by mythrocks** [comment]
Build

**Comment by mythrocks** [comment]
Build

**Comment by mythrocks** [comment]
Build

**Comment by mythrocks** [comment]
I've manually verified that this optimization works on Databricks.
The CI run seems to be failing with something unrelated to this functionality:
```
[2024-03-07T02:24:19.427Z] org.jenkinsci.plugins.scriptsecurity.sandbox.RejectedAccessException: Scripts not permitted to use method org.jenkinsci.plugins.workflow.support.steps.build.RunWrapper getRawBuild
[2024-03-07T02:24:19.427Z] 	at org.jenkinsci.plugins.scriptsecurity.sandbox.whitelists.StaticWhitelist.rejectMethod(StaticWhitelist.java:229)
[2024-03-07T02:24:19.427Z] 	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.rejectMethod(SandboxInterceptor.java:594)
[2024-03-07T02:24:19.427Z] 	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onMethodCall(SandboxInterceptor.java:167)
[2024-03-07T02:24:19.427Z] 	at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:178)
[2024-03-07T02:24:19.427Z] 	at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:182)
[2024-03-07T02:24:19.427Z] 	at com.cloudbees.groovy.cps.sandbox.SandboxInvoker.methodCall(SandboxInvoker.java:17)
[2024-03-07T02:24:19.427Z] 	at WorkflowScript.run(WorkflowScript:291)
[2024-03-07T02:24:19.427Z] 	at ___cps.transform___(Native Method)
...
[2024-03-07T02:24:19.427Z] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2024-03-07T02:24:19.427Z] 	at java.base/java.lang.Thread.run(Thread.java:829)
[2024-03-07T02:24:20.203Z] Finished: FAILURE
```

**Review by revans2 — APPROVED** [review]

**Comment by mythrocks** [comment]
Build

**Comment by NvTimLiu** [comment]
> 19.427

Looks like `getRawBuild` was removed from Jenkins ScriptApproval list by someone, added it back.

**Comment by NvTimLiu** [comment]
> I've manually verified that this optimization works on Databricks. The CI run seems to be failing with something unrelated to this functionality:
> 
> ```
> [2024-03-07T02:24:19.427Z] org.jenkinsci.plugins.scriptsecurity.sandbox.RejectedAccessException: Scripts not permitted to use method org.jenkinsci.plugins.workflow.support.steps.build.RunWrapper getRawBuild
> [2024-03-07T02:24:19.427Z] 	at org.jenkinsci.plugins.scriptsecurity.sandbox.whitelists.StaticWhitelist.rejectMethod(StaticWhitelist.java:229)
> [2024-03-07T02:24:19.427Z] 	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.rejectMethod(SandboxInterceptor.java:594)
> [2024-03-07T02:24:19.427Z] 	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onMethodCall(SandboxInterceptor.java:167)
> [2024-03-07T02:24:19.427Z] 	at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:178)
> [2024-03-07T02:24:19.427Z] 	at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:182)
> [2024-03-07T02:24:19.427Z] 	at com.cloudbees.groovy.cps.sandbox.SandboxInvoker.methodCall(SandboxInvoker.java:17)
> [2024-03-07T02:24:19.427Z] 	at WorkflowScript.run(WorkflowScript:291)
> [2024-03-07T02:24:19.427Z] 	at ___cps.transform___(Native Method)
> ...
> [2024-03-07T02:24:19.427Z] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
> [2024-03-07T02:24:19.427Z] 	at java.base/java.lang.Thread.run(Thread.java:829)
> [2024-03-07T02:24:20.203Z] Finished: FAILURE
> ```

Looks like `getRawBuild` was removed from Jenkins ScriptApproval list by someone, added it back. this `Scripts not permitted to use method org.jenkinsci.plugins.workflow.support.steps.build.RunWrapper getRawBuild` error should be not exist

**Pull request merged**